### PR TITLE
Added location-operations jests

### PR DESCRIPTION
**Pull request by ElemelonWind**
Wrote a few jests to test the call of `getUserLocation` in different cases. Also fixed a small typo in `location-operations` docs.

J=SLAP-2322
TEST=auto

### CODE REVIEW

File: docs/search-ui-react.getuserlocation.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 ## getUserLocation() function
 
-Retrieves user's location using nagivator.geolocation API.
```
Comment (oshi97) [inline docs/search-ui-react.getuserlocation.md:7]: nagivator

File: tests/operations/location-operations.test.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { SearchIntent, useSearchActions } from '@yext/search-headless-react';
+import { getUserLocation, updateLocationIfNeeded } from '../../src/utils/location-operations';
+
+describe('Location Operations updateLocationIfNeeded', () => {
+  it('fires getUserLocation with NearMe intent', async () => {
+    const searchActions = useSearchActions();
```
Comment (oshi97) [inline tests/operations/location-operations.test.tsx:6]: how is useSearchActions working outside of a react component here?

Comment (ElemelonWind) [reply tests/operations/location-operations.test.tsx:6]: uHHh honestly I'm not sure why it worked lmao

how would you suggest approaching this instead?

Comment (oshi97) [reply tests/operations/location-operations.test.tsx:6]: Would you mind digging a little into what's happening right now? If it works as is then that's good

**Comment by ElemelonWind** [comment]
Will probably add a test for second part of if statement (`!searchActions.state.location.userLocation`) tmrw

File: tests/operations/location-operations.test.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { SearchIntent } from '@yext/search-headless-react';
+import { generateMockedHeadless } from '../__fixtures__/search-headless';
+import { updateLocationIfNeeded } from '../__fixtures__/utils/location-operations';
+
+describe('Location Operations updateLocationIfNeeded', () => {
+
+  it('fires setUserLocation with NearMe intent', async () => {
+    const searchActions = generateMockedHeadless();
```
[the remark this answers is not in the document]
Comment (ElemelonWind) [reply tests/operations/location-operations.test.tsx:8]: I referenced Tom's [search-operations PR](https://github.com/yext/search-ui-react/commit/6e2d2f39930cf517484cebe7f772445b5885db2b) bc I figured it'd be the most similar to locations-operations. I don't think it makes a difference in this case?

[the remark this answers is not in the document]
Comment (oshi97) [reply tests/operations/location-operations.test.tsx:8]: Their purposes are a little different, `generateMockedHeadless()` makes a real AnswersHeadless instance using a mocked search-core (it doesn't make real requests) while the mock/spy methods are more lightweight. I think what we have here is fine!

**Review by oshi97 — APPROVED** [review]

**Pull request merged**